pkg/build/scripts/grub/guestfish-snippet.tpl

```diff
@@ -0,0 +1,13 @@
+# Configure GRUB for first boot
+# - Without this, the values wouldn't be used until after the first time the
+#   grub configuration is regenerated
+download /boot/grub2/grub.cfg /tmp/grub.cfg
+! sed -i '/ignition.platform/ s/$/ {{.KernelArgs}} /' /tmp/grub.cfg
+upload /tmp/grub.cfg /boot/grub2/grub.cfg
+
+# Configure GRUB defaults
+# - Without this, when `transactional-update grub.cfg` is run it will overwrite
+#   settings used in the above change
+download /etc/default/grub /tmp/grub
+! sed -i '/^GRUB_CMDLINE_LINUX_DEFAULT="/ s/"$/ {{.KernelArgs}} "/' /tmp/grub
```

Comment:
Would this not replace the `GRUB_CMDLINE_LINUX_DEFAULT` part as well? I've seen either repetitions or the use of `\1` to keep the matched string but none of those apply here.

Reply:
Sanity check me here, but it doesn't appear to. Here's the test I ran on the sed command:

`$ cat line-default
GRUB_CMDLINE_LINUX_DEFAULT="console=ttyS0,115200 console=tty0 security=selinux selinux=1 quiet net.ifnames=0 \$ignition_firstboot ignition.platform.id=metal"

$ sed '/^GRUB_CMDLINE_LINUX_DEFAULT="/ s/"$/ additional_option=xyz"/' ./line-default
GRUB_CMDLINE_LINUX_DEFAULT="console=ttyS0,115200 console=tty0 security=selinux selinux=1 quiet net.ifnames=0 \$ignition_firstboot ignition.platform.id=metal additional_option=xyz"
`

Reply:
Interesting, I didn't expect that but I'm not that good with sed in the first place. Thanks for the example!

Reply:
This sed command replaces the trailing `"` with ` {{.KernelArgs}} "`, my question is, are you sure you'll never have anything after the last `"` (i.e. no trailing spaces, no comments or anything) ?

Reply:
As far as we know right now. The sed approach may go away in the future, but looking at the context of that line, there's nothing that could be after the ending "